src/TelemetryChannels/ServerTelemetryChannel/Shared/AdaptiveSamplingTelemetryProcessor.cs

```diff
@@ -0,0 +1,262 @@
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets minimum sampling percentage that can be set 
+        /// by the dynamic sampling percentage algorithm.
+        /// </summary>
+        public double MinSamplingPercentage
+        {
+            get
+            {
+                return this.estimatorSettings.MinSamplingPercentage;
+            }
+
+            set
+            {
+                this.estimatorSettings.MinSamplingPercentage = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets maximum sampling percentage that can be set 
+        /// by the dynamic sampling percentage algorithm.
+        /// </summary>
+        public double MaxSamplingPercentage
+        {
+            get
+            {
+                return this.estimatorSettings.MaxSamplingPercentage;
+            }
+
+            set
+            {
+                this.estimatorSettings.MaxSamplingPercentage = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets duration of the sampling percentage evaluation 
+        /// interval in seconds.
+        /// </summary>
+        public int EvaluationIntervalSeconds
+        {
+            get
+            {
+                return this.estimatorSettings.EvaluationIntervalSeconds;
+            }
+
+            set
+            {
+                this.estimatorSettings.EvaluationIntervalSeconds = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating how long to not to decrease
+        /// sampling percentage after last change to prevent excessive fluctuation.
+        /// </summary>
+        public int SamplingPercentageDecreaseTimeoutSeconds
+        {
+            get
+            {
+                return this.estimatorSettings.SamplingPercentageDecreaseTimeoutSeconds;
+            }
+
+            set
+            {
+                this.estimatorSettings.SamplingPercentageDecreaseTimeoutSeconds = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating how long to not to increase
+        /// sampling percentage after last change to prevent excessive fluctuation.
+        /// </summary>
+        public int SamplingPercentageIncreaseTimeoutSeconds
+        {
+            get
+            {
+                return this.estimatorSettings.SamplingPercentageIncreaseTimeoutSeconds;
+            }
+
+            set
+            {
+                this.estimatorSettings.SamplingPercentageIncreaseTimeoutSeconds = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets exponential moving average ratio (factor) applied
+        /// during calculation of rate of telemetry items produced by the application.
+        /// </summary>
+        public double MovingAverageRatio
+        {
+            get
+            {
+                return this.estimatorSettings.MovingAverageRatio;
```

Comment:
when customer sets this you need to re-initialize ```ExponentialMovingAverageCounter```. When SDK reads configuration file - constructor will be called and construct everything with the standard settings. Only after that property will be overridden, but all the classes you've already initialized will not receive updated value